CONTRIBUTING.md

```diff
@@ -0,0 +1,56 @@
+# Contributing
```

Comment:
Great that you add this, but why is the file called `contributing.md`? I think it would make more sense to call it `architecture.md` or something similar.

Reply:
Changed the file name in the latest commit